### PR TITLE
add a provider parameter in SSLContextBuilder

### PR DESCRIPTION
**Pull request by lujianbo**
[HTTPCORE-450]
add a provider parameter in SSLContextBuilder to support third party provider

https://issues.apache.org/jira/browse/HTTPCORE-450

### CODE REVIEW

File: httpcore/src/main/java/org/apache/http/ssl/SSLContextBuilder.java [hunk 1]
```diff
@@ -272,6 +274,20 @@ public SSLContext build() throws NoSuchAlgorithmException, KeyManagementExceptio
         return sslcontext;
     }
 
+    public SSLContext buildWithProvider(final Provider provider) throws NoSuchAlgorithmException, KeyManagementException {
```
Comment (ok2c) [inline httpcore/src/main/java/org/apache/http/ssl/SSLContextBuilder.java]: Do it really take a special `#build` method? Can't we have a `#useProvider` setter similar to `#useProtocol` ?

Comment (lujianbo) [inline httpcore/src/main/java/org/apache/http/ssl/SSLContextBuilder.java]: buildWithProvider(final String provider) will throw a new NoSuchProviderException.

in order to be compatible with the old interface so add a new interface

Comment (ok2c) [inline httpcore/src/main/java/org/apache/http/ssl/SSLContextBuilder.java]: Fair point. However I am not sure this change should go into 4.4. In 5.0 we can change APIs as we deem fit.

Comment (lujianbo) [inline httpcore/src/main/java/org/apache/http/ssl/SSLContextBuilder.java]: i agree

**Pull request closed without merging**